Add general navigation components

lukitoki1 opened the pull request:
Add app-wide navigation components:
- side NavBar (with ability to shrinking and expanding),
- TopBar.

Resolves #67

Commit:
Add general navigation components

Add app-wide navigation components:
- side NavBar (with ability to shrinking and expanding),
- TopBar.

maciekkoz98 commented on frontend/compute-system/src/App.tsx:
```diff
@@ -1,7 +1,8 @@
-import { createBrowserHistory } from "history";
+import {createBrowserHistory} from "history";
```
use Prettier as code formatter. It does the job quite nice!

lukitoki1 replied:
I'm aware of the fact that this is a typical Prettier styling, but then Prettier should be added to dev dependencies imo.

maciekkoz98 replied:
how to add it then? ;p

lukitoki1 replied:
well, I'm working on it xd

lukitoki1 replied:
DONE

maciekkoz98 commented on frontend/compute-system/src/components/Navigation/NavDrawer/NavDrawer.tsx:
```diff
@@ -0,0 +1,66 @@
+      },
+    },
+  }),
+);
```
I would extract style lines into separate file

lukitoki1 replied:
DONE
DONE
DONE
DONE

maciekkoz98 commented on frontend/compute-system/src/components/Navigation/NavDrawer/TopBar.tsx:
```diff
@@ -0,0 +1,73 @@
+import {createStyles, makeStyles, Theme} from "@material-ui/core/styles";
+import {drawerWidth} from "../../../const/values";
+
+const useStyles = makeStyles((theme: Theme) =>
```
I would extract style lines to separate file

lukitoki1 replied:
DONE
DONE
DONE

maciekkoz98 commented on frontend/compute-system/src/App.tsx:
```diff
@@ -12,7 +13,9 @@ function App() {
         path="/"
         render={() => (
-          <AppStore />
+          <Navigation>
```
Can it be a parent of Router rather than AppStore?
Thanks to that every other route would have the drawer and the topbar

lukitoki1 replied:
Yep, good idea.

lukitoki1 replied:
DONE
DONE

Commit:
Add Prettier to dev dependencies and reformat PR

Commit:
Move styles to separate file

Commit:
Move Route inside Navigation component

maciekkoz98 commented on frontend/compute-system/src/components/Navigation/styles.ts:
```diff
@@ -0,0 +1,84 @@
+    },
+    toolbar: toolbarStyles(theme),
+  })
+);
```
breakline here

lukitoki1 replied:
DONEDONE

maciekkoz98 commented on frontend/compute-system/src/components/Navigation/styles.ts:
```diff
@@ -0,0 +1,84 @@
+      },
+    },
+  })
+);
```
breakline here

lukitoki1 replied:
DONEDONEDONEDONEDONEDONEDONEDONE

maciekkoz98 commented on frontend/compute-system/src/components/Navigation/styles.ts:
```diff
@@ -0,0 +1,84 @@
+      display: "none",
+    },
+  })
+);
```
breakline here

lukitoki1 replied:
DONEDONEDONEDONE

maciekkoz98 commented on frontend/compute-system/src/components/Navigation/styles.ts:
```diff
@@ -0,0 +1,84 @@
+
+export const rootStyles = () => {
+  return { display: "flex" };
+};
```
breakline here

lukitoki1 replied:
DONEDONEDONEDONEDONEDONE

maciekkoz98 reviewed (changes requested):
pierdolety left

Commit:
Add breaklines in styles declarations file

Commit:
Change naming convention for partial styles

Change naming convention for partial styles
to differentiate them from regular styles.
